packages/shared/src/SliderLabels.js

```diff
@@ -0,0 +1,148 @@
+// @flow
+
+import * as React from 'react';
+import { defaultTokens } from '@kiwicom/mobile-orbit';
+import { View } from 'react-native';
+import { Translation } from '@kiwicom/mobile-localization';
+
+import Text from './Text';
+import StyleSheet from './PlatformStyleSheet';
+
+type Props = {|
+  +startLabel: React.Element<typeof Translation>,
+  +endLabel?: React.Element<typeof Translation>,
+  +max: number,
+  +min: number,
+  +startValue: number,
+  +endValue?: number,
+|};
+
+type State = {|
+  width: number,
+  labelStartWidth: number,
+  labelEndWidth: number,
+  paddingLeft: number,
+  paddingRight: number,
+|};
+
+export default class SliderLabels extends React.Component<Props, State> {
+  constructor() {
+    super();
+    this.state = {
+      width: 0,
+      labelStartWidth: 0,
+      labelEndWidth: 0,
+      paddingLeft: 0,
+      paddingRight: 0,
+    };
+  }
+
+  componentDidMount = () => {
+    requestAnimationFrame(this.setPadding);
+  };
+
+  getMaxPadding = () => {
+    if (!this.props.endLabel) {
+      return Math.floor(this.state.width - this.state.labelStartWidth);
+    } else {
+      return Math.floor(
+        this.state.width -
+          this.state.labelStartWidth -
+          this.state.labelEndWidth,
+      );
+    }
+  };
+
+  calculateMarkerX = (type: 'start' | 'end'): number => {
+    const { min, max, startValue, endValue } = this.props;
+
+    if (type === 'start') {
+      return Math.round((startValue / (max - min)) * this.state.width);
+    } else if (type === 'end' && endValue) {
+      const val = (endValue / (max - min)) * this.state.width;
+      return Math.round(this.state.width - val);
+    } else {
+      return 0;
+    }
+  };
+
+  setPadding = () => {
+    const startMarkerX = this.calculateMarkerX('start');
+    const maxPadding = this.getMaxPadding();
+
+    const startLabelHalf = this.state.labelStartWidth / 2;
+    const startOffset =
+      startMarkerX < startLabelHalf ? 0 : startMarkerX - startLabelHalf;
+
+    if (!this.props.endLabel) {
+      this.setState({
+        paddingLeft: startOffset < maxPadding ? startOffset : maxPadding - 1,
+      });
+    } else {
+      const gap = 10;
+      const endMarkerX = this.calculateMarkerX('end');
+      const endLabelHalf = this.state.labelEndWidth / 2;
+      const endOffset =
+        endMarkerX < endLabelHalf ? 0 : endMarkerX - endLabelHalf;
+
+      if (startOffset + endOffset < maxPadding - gap) {
+        this.setState({
+          paddingLeft: startOffset,
+          paddingRight: endOffset,
+        });
+      }
+    }
+
+    requestAnimationFrame(this.setPadding);
+  };
+
+  saveFullWidth = (e: any) => {
```

Comment:
Please, don't use `any` as a type. There should be `type OnLayout` available in `@kiwicom/mobile-shared`.